src/path.rs

```diff
@@ -0,0 +1,44 @@
+use crate::grid::Grid;
+use pathfinding::prelude::bfs;
+
+pub fn resolve(grid: &Grid, goals: &[(i32, i32)]) -> Option<Vec<(i32, i32)>> {
+    let mut res = vec![];
+    for i in 1..goals.len() {
+        let prev_goal = goals[i - 1];
+        let goal = goals[i];
+        let to_add = bfs(
+            &prev_goal,
+            |p| {
```

Comment:
Please avoid these single letter names unless it's glaringly obvious what the binding is